lib/libyear_bundler/cli.rb

```diff
@@ -1,30 +1,28 @@
 require "bundler/cli"
 require "bundler/cli/outdated"
-require "libyear_bundler/report"
+require "libyear_bundler/options"
 require "libyear_bundler/query"
+require "libyear_bundler/report"
+
 
 module LibyearBundler
   # The `libyear-bundler` command line program
   class CLI
-    OPTIONS = %w[
-      --all
-      --grand-total
-      --libyears
-      --releases
-      --versions
-    ].freeze
-
     E_BUNDLE_OUTDATED_FAILED = 1
     E_NO_GEMFILE = 2
+    E_INVALID_CLI_ARG = 3
 
     def initialize(argv)
+      @options = ::LibyearBundler::Options.new(argv).parse
+      # Command line flags are removed form `argv` in `Options` by
+      # `OptionParser`, leaving non-flag command line arguments,
+      # such as a Gemfile path
```

Comment:
I get it now, thanks. That's a better comment. I didn't realize that `OptionParser` modifies `argv`.